### PR TITLE
Feature/google-auth

### PR DESCRIPTION
**Pull request by willarion**
saved user data from google auth with useReduser and provided it to the context to use it for user info display

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -2,8 +2,13 @@ import React from 'react';
 import { Logo } from '../Logo/Logo';
 import { HeaderBtn } from '../HeaderBtn/HeaderBtn';
 import { UserProfile } from '../UserProfile/UserProfile';
+import { CurrentUserContext } from '../../pages/HomePage/HomePage';
 
 export const Header: React.FC = () => {
+  const currentUser = React.useContext(CurrentUserContext);
+
+  console.log(currentUser);
```
Comment (AnastasiaKozhemyachko) [inline src/components/Header/Header.tsx]: Delete console.log(currentUser);

File: src/pages/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+
+export const CurrentUserContext = React.createContext<LoginState>(initialState);
 
 export const HomePage: React.FC = ({}) => {
+  const { userInfo, dispatchUserInfo } = useUserInfo(initialState);
+
+  const responseGoogle = (
+    response: GoogleLoginResponse | GoogleLoginResponseOffline,
+  ) => {
+    console.log(response);
```
Comment (AnastasiaKozhemyachko) [inline src/pages/HomePage/HomePage.tsx]: Delete console.log(response);

File: public/index.html [hunk 3]
```diff
@@ -1,44 +1,22 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <script src="https://apis.google.com/js/platform.js" async defer></script>
```
Comment (bondarchukse) [inline public/index.html]: you don't need to have this script with react-google-login library

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -2,8 +2,13 @@ import React from 'react';
 import { Logo } from '../Logo/Logo';
 import { HeaderBtn } from '../HeaderBtn/HeaderBtn';
 import { UserProfile } from '../UserProfile/UserProfile';
+import { CurrentUserContext } from '../../pages/HomePage/HomePage';
 
 export const Header: React.FC = () => {
+  const currentUser = React.useContext(CurrentUserContext);
```
Comment (bondarchukse) [inline src/components/Header/Header.tsx]: you will store more that only user profile in context. make sence to think about it now. You can provide custom hook to pick up user profile from store

````
const useUserProfile = (): UserPorfile | null => {
  const { user } = useContext(CurrentUserContext);
  return user;
}
````

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -18,7 +23,12 @@ export const Header: React.FC = () => {
           <div className="header-comp pull-right">
             <HeaderBtn to={'index.html'} buttonType={'icon-bell'} />
             <HeaderBtn to={'index.html'} buttonType={'icon-mail'} />
-            <UserProfile alt={'userpic'} to={'index.html'} />
+            <UserProfile
+              alt={'userpic'}
+              to={'index.html'}
```
Comment (bondarchukse) [inline src/components/Header/Header.tsx:26]: link hould reffer to the `'/'` url (home page)

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -18,7 +23,12 @@ export const Header: React.FC = () => {
...
+              name={currentUser.user?.name}
```
Comment (bondarchukse) [inline src/components/Header/Header.tsx]: I think would be better to have optional render for the user profile component. (render something else if user not signed in, login button for example)

File: src/components/UserProfile/UserProfile.tsx [hunk 7]
```diff
@@ -4,14 +4,20 @@ import { Link } from 'react-router-dom';
...
+  name,
+  src,
+}) => {
   return (
     <Link to={to} className="profile">
-      <span>User name</span>
-      {/*TODO get username and image? from context of user*/}
-      <img alt={alt} src="../../images/profilePhoto.png" />
+      <span>{name}</span>
+      <img alt={alt} src={src} />
```
Comment (bondarchukse) [inline src/components/UserProfile/UserProfile.tsx:20]: you can make 'alt' property optional

File: src/pages/HomePage/HomePage.tsx [hunk 8]
```diff
@@ -1,18 +1,66 @@
...
+} from 'react-google-login';
+import { UserProfile } from '../../models/UserProfile';
+
+export interface LoginState {
+  user: UserProfile | null;
+}
+
+const initialState: LoginState = { user: null };
+
+export const CurrentUserContext = React.createContext<LoginState>(initialState);
```
Comment (bondarchukse) [inline src/pages/HomePage/HomePage.tsx]: `CurrentUserContext` - would be better to name it as `LoginContext` or somethig else. You should store other loging information here

File: src/pages/HomePage/HomePage.tsx [hunk 9]
```diff
@@ -1,18 +1,66 @@
...
 
 export const HomePage: React.FC = ({}) => {
+  const { userInfo, dispatchUserInfo } = useUserInfo(initialState);
+
+  const responseGoogle = (
```
Comment (bondarchukse) [inline src/pages/HomePage/HomePage.tsx:28]: you forgot about error handling, try to open sign in google window and then close it without confirmation for example.

Comment (willarion) [reply src/pages/HomePage/HomePage.tsx:28]: I use ` onFailure={(response) => console.log(response)}` at string 46. I should put it inside responseGoogle function, rigth? @bondarchukse

Comment (bondarchukse) [reply src/pages/HomePage/HomePage.tsx:28]: it's ok to set error message to the console for now, but needs to add TODO comment to handle this case in future

File: src/pages/HomePage/HomePage.tsx [hunk 10]
```diff
@@ -1,18 +1,71 @@
...
+
   return (
     <section>
-      <Header />
-      <Menu />
-      <ContainerMain>
-        <Submenu />
-      </ContainerMain>
+      <GoogleLogin
+        clientId="481156142014-iqr96oii4rvkk5og1eglruv6pdujkof4.apps.googleusercontent.com"
```
Comment (AnastasiaKozhemyachko) [inline src/pages/HomePage/HomePage.tsx:43]: if it's the same, create a constant for it

File: src/components/Header/Header.tsx [hunk 11]
```diff
@@ -16,9 +19,18 @@ export const Header: React.FC = () => {
...
+            <HeaderBtn to={'/'} buttonType={'icon-bell'} />
+            <HeaderBtn to={'/'} buttonType={'icon-mail'} />
+            <UserProfile
+              alt={'userpic'}
+              to={'index.html'}
+              name={user?.name}
+              src={
+                user
+                  ? user.imageUrl
+                  : 'https://www.worldfuturecouncil.org/wp-content/uploads/2020/06/blank-profile-picture-973460_1280-1-300x300.png'
```
Comment (AnastasiaKozhemyachko) [inline src/components/Header/Header.tsx:31]: you can put it in a separate function for better readability

**Pull request merged**